src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -591,7 +591,32 @@ private function getAdjacentRecordID($offset)
 
         $map = $list->limit($limit, $limitOffset)->column('ID');
         $index = array_search($this->record->ID, $map ?? []);
-        return isset($map[$index+$offset]) ? $map[$index+$offset] : false;
+        $position = $index + $offset;
+        $page = $currentPage;
+        $lastPosition = count($map) - 1;
+        $hasMorePages = $this->getNumPages($gridField) > $currentPage;
+
+        if ($position === 0 && $currentPage > 1) {
+            $page = $currentPage - 1;
+        } elseif ($position === $lastPosition && $hasMorePages) {
+            $page = $currentPage + 1;
+        }
+        $gridField->State->GridFieldPaginator->currentPage = (int)$page;
+        
+        return isset($map[$position]) ? $map[$position] : false;
+    }
+
+    /**
+     * Gets the number of GridField pages
+     * @return int
+     */
+    public function getNumPages(GridField $gridField)
```

Comment:
```suggestion
    private function getNumPages(GridField $gridField)
```
No need for this to be public.

Reply:
DONE